daemon/algod/api/server/router.go

```diff
@@ -104,14 +113,18 @@ func NewRouter(logger logging.Logger, node *node.AlgorandFullNode, shutdown <-ch
 
 	// Registering v2 routes
 	v2Handler := v2.Handlers{
-		Node:     apiNode{node},
+		Node:     node,
 		Log:      logger,
 		Shutdown: shutdown,
 	}
 	nppublic.RegisterHandlers(e, &v2Handler, apiAuthenticator)
 	npprivate.RegisterHandlers(e, &v2Handler, adminAuthenticator)
-	ppublic.RegisterHandlers(e, &v2Handler, apiAuthenticator)
-	pprivate.RegisterHandlers(e, &v2Handler, adminAuthenticator)
+	if node.Config().NodeSyncMode {
+		data.RegisterHandlers(e, &v2Handler, apiAuthenticator)
+	} else {
+		ppublic.RegisterHandlers(e, &v2Handler, apiAuthenticator)
+		pprivate.RegisterHandlers(e, &v2Handler, adminAuthenticator)
```

Comment:
I don't think this needs to be conditional becaues the data node returns an error. Is there another benefit to skipping RegisterHandlers?

Reply:
I think the main thing is reducing the possibility of hitting a NPE or similar in the node because of uninitialized values somewhere.